Config system improvements: compositional config, added scenes_dir, removed config_dir, fixed eqa test

mathfac opened the pull request:
- https://github.com/facebookresearch/habitat-api/issues/63, https://github.com/facebookresearch/habitat-api/issues/25: Added `scenes_dir` parameter to Config to have no requirement of `data` symlink or data directory in working directory. `scenes_dir` will be used as prefix for `scene_id` during  episodes dataset loading.
- Added overwriting options argument to `get_config` that unblocks parameter sweeping and experimentation capabilities from command line
- https://github.com/facebookresearch/habitat-api/issues/29: `config_file` path can be comma separated list of configs that will be merged in exact order. That will help to use separate configs for dataset specification and sensors setups. String with commas was chosen because that make it easy to pass list of configs as script argument.
- To make habitat-api easier to as package we decided to get rid of forcing loading of configs from `config_dir` param
- https://github.com/facebookresearch/habitat-api/issues/75: fixed unit test caused by latest habitat-sim update

**Caution:** with this change all config paths will be specified in new way: `tasks/poingoal.yaml` => `configs/tasks/pointgoal.yaml` 

## How Has This Been Tested
`pytest`
`black . --line-length 79;mypy . --ignore-missing-imports;pycodestyle;flake8`
```
============================================================================================================ test session starts =============================================================================================================
platform linux -- Python 3.6.4, pytest-4.4.1, py-1.5.2, pluggy-0.9.0
rootdir: /private/home/maksymets/habitat-api, inifile: setup.cfg, testpaths: test
plugins: xdist-1.25.0, forked-0.2
collected 37 items

test/test_baseline_agents.py ..                                                                                                                                                                                                        [  5%]
test/test_config.py .                                                                                                                                                                                                                  [  8%]
test/test_dataset.py .........                                                                                                                                                                                                         [ 32%]
test/test_habitat_env.py .........s                                                                                                                                                                                                    [ 59%]
test/test_habitat_example.py ...                                                                                                                                                                                                       [ 67%]
test/test_install.py .                                                                                                                                                                                                                 [ 70%]
test/test_mp3d_eqa.py ...                                                                                                                                                                                                              [ 78%]
test/test_pointnav_dataset.py ...                                                                                                                                                                                                      [ 86%]
test/test_sensors.py ...                                                                                                                                                                                                               [ 94%]
test/test_trajectory_sim.py ..                                                                                                                                                                                                         [100%]
========================================================================================================== short test summary info ===========================================================================================================
SKIPPED [1] test/test_habitat_env.py:318: unconditional skip

=================================================================================================== 36 passed, 1 skipped in 210.82 seconds ===================================================================================================
```
<!--- Please describe here how your modifications have been tested. -->

## Types of changes

<!--- What types of changes does your code introduce? Leave all the items that apply: -->
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

erikwijmans commented on habitat/datasets/pointnav/pointnav_dataset.py:
```diff
@@ -96,15 +97,28 @@ def __init__(self, config: Config = None) -> None:
+                episode.scene_id = episode.scene_id[
+                    len(DEFAULT_SCENE_PATH_PREFIX) :
+                ]
+                episode.scene_id = os.path.join(scenes_dir, episode.scene_id)
```
Should this join be outside the if?  That'll give us flexibility in the future to not have all scene paths start with `data/`

erikwijmans commented on test/test_mp3d_eqa.py:
```diff
@@ -193,7 +193,7 @@ def test_mp3d_eqa_sim_correspondence():
                 "".format(
                     cur_state.position - point.position,
-                    cur_state.rotation - point.rotation,
+                    np.array(cur_state.rotation) - point.rotation,
```
This'll need to be something like https://github.com/facebookresearch/habitat-sim/blob/master/habitat_sim/utils.py#L27 instead of just `np.array` as we have been storing quats in `[x, y, z, w]` form instead of the standard `[w, x, y, z]` form because that's how Eigen does it.

mathfac commented on baselines/config/default.py:
```diff
@@ -60,10 +60,15 @@
-def cfg(
-    config_file: Optional[str] = None, config_dir: str = DEFAULT_CONFIG_DIR
-) -> CN:
+def get_config(config_file: Optional[str] = None) -> CN:
```
@erikwijmans thank you for the suggestions. I adopted it in some way. Regarding `*args: str` that doesn't play well with overwriting options `opts` I added to make experimentations easier. That's almost pythonic way.

erikwijmans replied:
Yeah, supporting override opts is probably more valuable than my desire for `*args: str` to support my laziness :)

mathfac commented:
Addressed the comments. Added support of config list, corrected scene_path, added overwriting options, added EQA dataset changes, adopted quaternion suggestion.
